kusto/kcsb_test.go

```diff
@@ -0,0 +1,74 @@
+package kusto
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildConnectionStringWithEnv(t *testing.T) {
+	ecsb := &ConnectionStringBuilder{}
+	ecsb.clusterURI = "endpoint"
+	ecsb.envAuth = true
+	csb, err := BuildConnectionStringWithEnv("endpoint")

```

Comment:
may be a table driven test ?

Define a HigherOrderFunction that returns csb,err ? and have a single test function that tests scenarios ?